app/models/notification.rb

```diff
@@ -137,7 +137,17 @@ def update_notifications(notifiable, action = nil)
     end
 
     def fast_destroy_old_notifications(destroy_before_timestamp = 4.months.ago)
-      Notification.where("created_at < ?", destroy_before_timestamp).in_batches(of: 10_000).delete_all
+      notification_sql = <<-SQL
+        DELETE FROM notifications
+        WHERE notifications.id IN (
+          SELECT notifications.id
+          FROM notifications
+          WHERE created_at < '#{destroy_before_timestamp}'
+          LIMIT 50000
+        )
+      SQL
```

Comment:
To establish a good pattern in the code like this I think we should use `sanitize_sql` here, something like:

```ruby
notification_sql = <<-SQL
  DELETE FROM notifications
  WHERE notifications.id IN (
    SELECT notifications.id
    FROM notifications
    WHERE created_at < ?
    LIMIT 50000
  )
SQL

notification_sql = Notification.sanitize_sql([notification_sql, destroy_before_timestamp])
```

everytime we mix SQL with variables we should keep this in mind